File: pagamento-semplice/php/esito.php
                                            
<?php

// Pagamento semplice - Esito

// Chiave segreta 
$CHIAVESEGRETA = "<CHIAVE SEGRETA PER CALCOLO MAC>"; // Sostituire con il valore fornito da Nexi

// Controllo che ci siano tutti i parametri di ritorno obbligatori per calcolare il MAC
$requiredParams = array('codTrans', 'esito', 'importo', 'divisa', 'data', 'orario', 'codAut', 'mac');
foreach ($requiredParams as $param) {
    if (!isset($_REQUEST[$param])) {
        echo 'Paramentro mancante ' . $param;
        exit;
    }
}

// Calcolo MAC con i parametri di ritorno
$macCalculated = sha1('codTrans=' . $_REQUEST['codTrans'] .
        'esito=' . $_REQUEST['esito'] .
        'importo=' . $_REQUEST['importo'] .
        'divisa=' . $_REQUEST['divisa'] .
        'data=' . $_REQUEST['data'] .
        'orario=' . $_REQUEST['orario'] .
        'codAut=' . $_REQUEST['codAut'] .
        $CHIAVESEGRETA
);

// Verifico corrispondenza tra MAC calcolato e parametro mac di ritorno
if ($macCalculated != $_REQUEST['mac']) {
    echo 'Errore MAC: ' . $macCalculated . ' non corrisponde a ' . $_REQUEST['mac'];
    exit;
}

// Nel caso in cui non ci siano errori gestisco il parametro esito
if ($_REQUEST['esito'] == 'OK') {
    echo 'La transazione ' . $_REQUEST['codTrans'] . " è avvenuta con successo; codice autorizzazione: " . $_REQUEST['codAut'];
} else {
    echo 'La transazione ' . $_REQUEST['codTrans'] . " è stata rifiutata; descrizione errore: " . $_REQUEST['messaggio'];
}
                    
                

File: pagamento-ricorrente/dotnet/esito.cs
                                            
// Pagamento ricorrente - Primo pagamento - Esito

using System;
using System.Collections.Generic;
using System.Text;
using System.Security.Cryptography;

namespace XPay
{
    class SampleXPayResponse
    {
        static void Main(string[] args)
        {
            // Chiave segreta
            string chiaveSegreta = "<CHIAVE SEGRETA PER CALCOLO MAC>"; // Sostituire con il valore fornito da Nexi

            // Calcolo MAC
            SampleXPayRequest sampleXPay = new SampleXPayRequest();
            string mac = sampleXPay.HashMac(  "codTrans=" + args.codTrans +
                                              "esito=" + args.esito +
                                              "importo=" + args.importo +
                                              "divisa=" + args.divisa +
                                              "data=" + args.data +
                                              "orario=" + args.orario +
                                              "codAut=" + args.codAut +
                                              chiaveSegreta);

            // Verifico corrispondenza tra MAC calcolato e MAC di ritorno
            if(mac != args.mac)
            {
                Console.WriteLine("errore MAC: " + mac + " NON CORRISPONDENTE A " + args.mac);
            } else
            {
                // Nel caso in cui non ci siano errori gestisco il parametro esito
                if (args.esito == "OK")
                {
                    Console.WriteLine("OK, pagamento avvenuto, preso riscontro");
                } else
                {
                    Console.WriteLine("KO, pagamento non avvenuto, preso riscontro");
                }
            }


        }
    }
}
                    
                